protocol/client/contact.go

```diff
@@ -56,6 +58,32 @@ type Contact struct {
 	PublicKey *ecdsa.PublicKey `json:"-"`
 }
 
+// CreateContactPrivate creates a new private contact.
+func CreateContactPrivate(name, pubKeyHex string) (c Contact, err error) {
+	pubKeyBytes, err := hexutil.Decode(pubKeyHex)
+	if err != nil {
+		return
+	}
+
+	c.Name = name
+	c.Type = ContactPrivate
+	c.State = ContactAdded
+	c.Topic = DefaultPrivateTopic()
+	c.PublicKey, err = crypto.UnmarshalPubkey(pubKeyBytes)
```

Comment:
i would rather return nil instead of contact if there was an error

Reply:
We return a value not a pointer here so `nil` is not an option. However, whenever an error is returned, one should never rely on the value returned along. I think that's obvious and "official" Go recommendation.